roles/client_side_tests/tasks/get_prom_info.yml

```diff
@@ -1,14 +1,47 @@
 ---
-- name: "Get the default-prometheus-htpasswd secret"
-  ansible.builtin.shell:
-    cmd: |
-      oc  get secret default-prometheus-htpasswd -ojson | jq '.data.auth, .data.password' |  sed 's/"//g'
-  register: prom_secret
-  changed_when: false
+- when: prom_auth_method == 'password'
+  block:
+  - name: "Get the default-prometheus-htpasswd secret"
```

Comment:
This diff is unfair; every one of these blocks was in the existing code! ;) The changes are mostly indenting.